### PR TITLE
Added Issue Templates and Refactored some code.

### PR DESCRIPTION
**Pull request by Sai-Suraj-27**
Added ISSUE TEMPLATES to fix issue #13 and refactored some code to make it better and more readable.

### CODE REVIEW

**Comment by Sai-Suraj-27** [comment]
@SagiPolaczek sir, Can you review this?

**Comment by SagiPolaczek** [comment]
@Sai-Suraj-27  Sure! Sorry for the delay, I missed the notification.

Also I'll take this opportunity to test some tool I encountered lately: @CodiumAI-Agent

File: fusedrug/data/molecule/smi_character_histogram.py [hunk 1]
```diff
@@ -79,7 +79,7 @@ def smi_file_character_histogram_multiprocessed(
         as_iterator=True,
         copy_to_global_storage=dict(indexed_text_file=itf),
     ):
-        total_characters_counter.update(curr_counter)
+        total_characters_counter |= curr_counter
```
Comment (SagiPolaczek) [inline fusedrug/data/molecule/smi_character_histogram.py:82]: Although both are equivalent lets stay with `.update()` for better readability by the reader

File: .github/workflows/ISSUE_TEMPLATE/---bug-report.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+---
```
Comment (SagiPolaczek) [inline .github/workflows/ISSUE_TEMPLATE/---bug-report.md:1]: Let's proceed with moving the templates under the `.github/ISSUE_TEMPLATE` directory, please. Just to be consistent with other projects.

**Review by SagiPolaczek — CHANGES_REQUESTED** [review]
Thanks @Sai-Suraj-27 for your contribution!  😄 

Please see two minors inlines

**Review by SagiPolaczek — APPROVED** [review]
Thanks! LGTM!

**Comment by SagiPolaczek** [comment]
Closes #13

**Pull request merged**